features/steps/steps.py

```diff
@@ -168,12 +167,15 @@ def when_i_configure_uaclient_using_proxy_machine(
         DEFAULT_CONFIG_FILE, tmp_local_conf
     )
     cfg = yaml.safe_load(load_file(tmp_local_conf))
+    new_cfg = {}
 
     if "ua_config" not in cfg:
-        cfg["ua_config"] = {}
-    cfg["ua_config"][proxy_cfg + "_proxy"] = proxy_cfg_value
+        new_cfg["ua_config"] = {}
+        new_cfg["ua_config"][proxy_cfg + "_proxy"] = proxy_cfg_value
+        context.text = yaml.dump(new_cfg).strip()
+    else:
+        context.text = "  {}: {}".format(proxy_cfg + "_proxy", proxy_cfg_value)
```

Comment:
This line assumes that the `ua_config` key is at the end of the file. We may need something like this instead:
```suggestion
        new_cfg["ua_config"] = cfg["ua_config"]
        new_cfg["ua_config"][proxy_cfg + "_proxy"] = proxy_cfg_value
        context.text = yaml.dump(new_cfg).strip()
```

Reply:
Yep, you are right on that one. But this code will be problematic because on that situation, we will create two entries of `ua_config` on the `uaclient.conf` file.

An other thing we could do here is rewrite the config instead of appending. In that situation, the original code should work fine. What do you think ?

Reply:
> we will create two entries of ua_config on the uaclient.conf file
My confusion stems from not expecting that to be a problem :upside_down_face: - I would expect the yaml parser to overwrite the previous value, but then again I didn't actually look at the yaml spec or test it out to see what pyyaml actually does:grimacing: 
> rewrite the config instead of appending
Yeah thats probably the best solution +1. Note that this is what I started doing and ran into permissions problems, which is why I ended up just calling the append step. You'll likely have to do something like:
```
    cmd = "printf '{}\n' > {}".format(context.text, DEFAULT_CONFIG_FILE)
    cmd = 'sh -c "{}"'.format(cmd)
    when_i_run_command(context, cmd, "with sudo")
```

Reply:
Tbh, the tests were failing after the rebase and when I looked at the `uaclient.conf` I assumed that this was the problem. Let me double check that and confirm if they do fail.

Maybe I was executing the tests with wrong parameters and assumed the error was caused by this.

Reply:
@orndorffgrant You are right, that was not the problem on the test. I have removed that change.

However, I still think the best way is to rewrite the config, but we can do this in another PR